query/search.go

```diff
@@ -0,0 +1,105 @@
+package query
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type searchRequest struct {
+	Term                string
+	From                int
+	Size                int
+	Types               []string
+	Index               string
+	Queries             []string
+	SortBy              string
+	AggregationField    string
+	Highlight           bool
+	FilterOnLatest      bool
+	FilterOnFirstLetter string
+	ReleasedAfter       string
+	ReleasedBefore      string
+	UriPrefix           string
+	Topic               []string
+	TopicWildcard       []string
+	Upcoming            bool
+	Published           bool
+	Now                 string
+}
+
+// SetupSearch loads templates for use by the search handler and should be done only once
+func SetupSearch() (*template.Template, error) {
+	//Load the templates once, the main entry point for the templates is search.tmpl. The search.tmpl takes
+	//the SearchRequest struct and uses the Request to build up the multi-query queries that is used to query elastic.
+	templates, err := template.ParseFiles(
+		"templates/search/search.tmpl",
```

Comment:
Probably, I am not sure this is the best way to be initialising the templates anyway but for this story I wanted to leave the query generation code as untouched as possible. I suggest that when we get into actually amending the functionality of the search rather than the 'plumbing' we'll want to look at this in a lot more detail. At the moment it's just been moved from one package to another as-is.

Reply:
Story raised in backlog to address this.